src/pytorch_lightning/strategies/launchers/spawn.py

```diff
@@ -41,21 +42,30 @@ class _SpawnLauncher(_Launcher):
     Note:
         - This launcher requires all objects to be pickleable.
         - It is important that the entry point to the program/script is guarded by ``if __name__ == "__main__"``.
+        - With start method 'fork' the user must ensure that no CUDA context gets created in the main process before
+          the launcher is invoked. E.g., one should avoid creating cuda tensors or calling ``torch.cuda.*`` functions
+          before calling ``Trainer.fit``.
 
     Args:
         strategy: A reference to the strategy that is used together with this launcher.
+        start_method: The method how to start the processes.
+            - 'spawn': The default start method. Requires all objects to be pickleable.
+            - 'fork': Preferrable for IPython/Jupyter environments where 'spawn' is not available. Not available on
+              the Windows platform.
     """
 
-    def __init__(self, strategy: Strategy) -> None:
+    def __init__(self, strategy: Strategy, start_method: Literal["spawn", "fork"] = "spawn") -> None:
         self._strategy = strategy
-        self._start_method = "spawn"
+        self._start_method = start_method
+        if start_method == "fork" and not hasattr(os, "fork"):
+            raise ValueError("The start method 'fork' is not available on this platform. Use 'spawn' instead.")
```

Comment:
@awaelchli Is there a case where this error would be thrown?
As we are validating it in Accelerator Connector and the error would be thrown there

Reply:
@justusschock It does check for platform compatibly in Accelerator Connector
https://github.com/Lightning-AI/lightning/blob/feature/ddp-fork2/src/pytorch_lightning/trainer/connectors/accelerator_connector.py#L616

Reply:
Didn't see that. But theoretically the launcher could be reused for other third-party plugins as well, so I think it is good to error out here again